src/app/components/HoldRequest/HoldRequest.jsx

```diff
@@ -14,19 +15,42 @@ class HoldRequest extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state = {
-      data: Store.getState(),
-      patron: PatronStore.getState(),
-    };
+    this.state = { patron: PatronStore.getState() };
+
+    // change all the components :(
     this.onChange = this.onChange.bind(this);
+    this.submitRequest = this.submitRequest.bind(this);
   }
 
   componentDidMount() {
     this.requireUser();
   }
 
   onChange() {
-    this.setState({ data: Store.getState() });
+    this.setState({ patron: PatronStore.getState() });
```

Comment:
Correct about the `Action`, but you don't need to import Actions if you have a store. They can be separate.

For the update on `PatronStore`, that's right. But keeping it for now. The other refactor of making it props is a bigger change/update that affects many components. So just kept this simple.